sql/sql_select.cc

```diff
@@ -1895,7 +1897,27 @@ int JOIN::init_join_caches()
     {
       table->prepare_for_keyread(tab->index, table->read_set);
     }
-    if (tab->cache && tab->cache->init(select_options & SELECT_DESCRIBE))
+
+    /*
+      We are about to initialize a join buffer.
+      Can we use lightweight initalization mode just for EXPLAINs? We can if
+      we're certain that the optimizer will not execute the subquery.
```

Comment:
How the term `subquery` gets into this context? We are looping through tables of the JOIN, how are subqueries related to them?